src/librustc/lint/levels.rs

```diff
@@ -247,17 +248,36 @@ impl<'a> LintLevelsBuilder<'a> {
                         );
                     }
 
-                    continue
-                }
+                    if gate_feature || !known_tool {
+                        continue
+                    }
+
+                    Some(lint_tool.as_str())
+                } else {
+                    None
+                };
                 let name = word.name();
-                match store.check_lint_name(&name.as_str()) {
+                match store.check_lint_name(&name.as_str(), tool_name) {
                     CheckLintNameResult::Ok(ids) => {
                         let src = LintSource::Node(name, li.span);
                         for id in ids {
                             specs.insert(*id, (level, src));
                         }
                     }
 
+                    CheckLintNameResult::Tool(result) => {
+                        if let Some(ids) = result {
+                            let complete_name = &format!("{}::{}", tool_name.unwrap(), name);
+                            let src = LintSource::Node(Symbol::intern(complete_name), li.span);
+                            for id in ids {
+                                specs.insert(*id, (level, src));
+                            }
+                        }
+                        //FIXME: if Tool(None) is returned than the lint either does not exist in
+                        //the lint tool or the code doesn't get compiled with the lint tool and
+                        //therefore the lint cannot exist.
```

Comment:
I'm not sure what to do here... Can we somehow detect if we currently use the `clippy_driver` or the `rustc_driver` to find out if the lint does not exists or wasn't even added in the first place?

Reply:
The driver could register itself as a tool. Not sure where that information belongs, maybe a new field in the `Session`?

Reply:
I think tool attributes are required to be checked by the tool. So maybe instead we leave the "unknown lint" error to clippy?

Reply:
Leaving it to the tool/Clippy was also my idea. We basically need a lint, which looks for `allow(clippy::_)`/... and checks if the lint exists, I think.

Reply:
So... I guess just turn the FIXME comment into a comment explaining that

Reply:
Ok will do. And I will squash these commits while doing this.